src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ImperialFirePatch.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using HarmonyLib;
+using Helpers;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches.Perks.Intelligence.Engineering {
+
+  public class ImperialFirePatch : PatchBase<ImperialFirePatch> {
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo AttackerTargetMethodInfo =
+      Type.GetType("SandBox.ViewModelCollection.MapSiege.MapSiegeProductionVM, SandBox.ViewModelCollection, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null")?
```

Comment:
This makes the mod dependent on Sandbox. Add `SandBox.ViewModelCollection.dll` to the assembly dependencies and use it to get the type instead of strings.
@Tyler-IN is there any problem with us depending on SandBox? For the load order stuff.

Reply:
Using Type.GetType will work for always-loaded deps like this.

If the type init fails because Sandbox isn't loaded, the patch will be skipped.

This keeps loose coupling. It's okay.